cmd/server-main.go

```diff
@@ -80,7 +80,7 @@ EXAMPLES:
 // Check for updates and print a notification message
 func checkUpdate(mode string) {
 	// Its OK to ignore any errors during getUpdateInfo() here.
-	if older, downloadURL, err := getUpdateInfo(1*time.Second, mode); err == nil {
+	if older, downloadURL, err := getUpdateInfo(mode, 1*time.Second); err == nil {
```

Comment:
@balamurugana This PR is only trying to improve the understandability of the code. It is not trying to fix any bugs. Let me try to list the improvements that are being attempted.

The struct is helping for the following reasons: 

### Avoid threading the `mode` argument through 5 functions

Excerpts from existing code:

```go
func getUpdateInfo(timeout time.Duration, mode string) (older time.Duration, downloadURL string, err error) {...}
func getLatestReleaseTime(timeout time.Duration, mode string) (releaseTime time.Time, err error) {...}
func DownloadReleaseData(timeout time.Duration, mode string) (data string, err error) {...}
func downloadReleaseData(releaseChecksumURL string, timeout time.Duration, mode string) (data string, err error) {...}
func getUserAgent(mode string) string {...}
```

Guess how many of the above functions actually use `mode` than just pass it around?

Just one.


### Avoid trivial functions to just hide and pass-in arguments to functions

1. `IsSourceBuild()`
```go
func isSourceBuild(minioVersion string) bool {
	_, err := time.Parse(time.RFC3339, minioVersion)
	return err != nil
}

func IsSourceBuild() bool {
	return isSourceBuild(Version)
}
```

2. `GetCurrentReleaseTime()`

```go
func GetCurrentReleaseTime() (releaseTime time.Time, err error) {
	return getCurrentReleaseTime(Version, os.Args[0])
}
```

3. `DownloadReleaseData()`

```go
func DownloadReleaseData(timeout time.Duration, mode string) (data string, err error) {
	return downloadReleaseData(minioReleaseURL+"minio.shasum", timeout, mode)
}

```

In this change, I am eliminating these multiple similarly named functions.


### Remove excessive string concatenation

In addition to changes related to introducing the struct, `getUserAgent()` is refactored to be significantly more readable (and less importantly, to have better performance).

```go
// Old
func getUserAgent(mode string) string {
	userAgent := "Minio (" + runtime.GOOS + "; " + runtime.GOARCH
	if mode != "" {
		userAgent += "; " + mode
	}
	if IsDocker() {
		userAgent += "; docker"
	}
	if IsSourceBuild() {
		userAgent += "; source"
	}
	userAgent += ") " + " Minio/" + Version + " Minio/" + ReleaseTag + " Minio/" + CommitID

	return userAgent
}

// New
func (u *updateCheckParams) getUserAgent() string {
	envTokens := []string{runtime.GOOS, runtime.GOARCH}
	if u.serverMode != "" {
		envTokens = append(envTokens, u.serverMode)
	}
	if u.isDocker() {
		envTokens = append(envTokens, "docker")
	}
	if u.isSourceBuild() {
		envTokens = append(envTokens, "source")
	}
	envInfo := strings.Join(envTokens, "; ")
	return fmt.Sprintf("Minio (%s)  Minio/%s Minio/%s Minio/%s",
		envInfo, Version, ReleaseTag, CommitID)
}
```

I hope the points above better illuminate the improvements. They are already mentioned in short in the commit message. I hope your argument is not that these changes are too trivial to be included.